sdcm/nemesis.py

```diff
@@ -859,6 +860,94 @@ def disrupt_snapshot_operations(self):
         if result.stderr:
             self.tester.fail(result.stderr)
 
+    def disrupt_show_toppartitions(self):
+        def _parse_toppartitions_output(output):
+            """parsing output of toppartitions
+
+            input format stored in output parameter:
+            WRITES Sampler:
+              Cardinality: ~10 (15 capacity)
+              Top 10 partitions:
+                Partition     Count       +/-
+                9        11         0
+                0         1         0
+                1         1         0
+
+            READS Sampler:
+              Cardinality: ~10 (256 capacity)
+              Top 3 partitions:
+                Partition     Count       +/-
+                0         3         0
+                1         3         0
+                2         3         0
+                3         2         0
+
+            return Dict:
+            {
+                'READS': {
+                    'toppartitions': '10',
+                    'partitions': OrderedDict('0': {'count': '1', 'margin': '0'},
+                                              '1': {'count': '1', 'margin': '0'},
+                                              '2': {'count': '1', 'margin': '0'}),
+                    'cardinality': '10',
+                    'capacity': '256',
+                },
+                'WRITES': {
+                    'toppartitions': '10',
+                    'partitions': OrderedDict('10': {'count': '1', 'margin': '0'},
+                                              '11': {'count': '1', 'margin': '0'},
+                                              '21': {'count': '1', 'margin': '0'}),
+                    'cardinality': '10',
+                    'capacity': '256',
+                    'sampler': 'WRITES'
+                }
+            }
+
+
+            Arguments:
+                output {str} -- stdout of nodetool topparitions command
+
+            Returns:
+                dict -- result of parsing
+            """
+
+            pattern1 = "(?P<sampler>[A-Z]+)\sSampler:\W+Cardinality:\s~(?P<cardinality>[0-9]+)\s\((?P<capacity>[0-9]+)\scapacity\)\W+Top\s(?P<toppartitions>[0-9]+)\spartitions:"
+            pattern2 = "(?P<partition>[\w:]+)\s+(?P<count>[\d]+)\s+(?P<margin>[\d]+)"
+            toppartitions = {}
+            for out in output.split('\n\n'):
+                partition = OrderedDict()
+                sampler_data = re.match(pattern1, out, re.MULTILINE)
+                sampler_data = sampler_data.groupdict()
+                partitions = re.findall(pattern2, out, re.MULTILINE)
+                for v in partitions:
+                    partition.update({v[0]: {'count': v[1], 'margin': v[2]}})
+                sampler_data.update({'partitions': partition})
+                toppartitions[sampler_data.pop('sampler')] = sampler_data
+            return toppartitions
+
+        self._set_current_disruption("ShowTopPartitions")
+        self.target_node.run_nodetool('cfstats')
```

Comment:
add a comment that we run cfstats because of the bug in Scylla

Reply:
added

Reply:
Fixed in branch next-3.1